### PR TITLE
aws2-kinesis: Make Kinesis KCL application name configurable

### PR DESCRIPTION
**Pull request by mikey-austin**
# Description

tl;dr
- make KCL application name configurable
- allow a custom _async_ kinesis client instance to be configured

The KCL application name is important to define, as the underlying AWS Kinesis library will use this application name as the basis for the dynamodb lease table(s) that it manages. My use case required the Kinesis stream name to differ from the dynamodb lease table name, and this was not possible due to the streamName being hard-coded to equal the applicationName when instantiating the AWS KCL consumer.

Also allow a pre-constructed async Kinesis client instance to be configured in the same manner as the non-async Kinesis client. My use case requires a custom AWS credentials provider, which meant that it was not possible before to use the existing async client builder capability in this module.

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

### CODE REVIEW

File: components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java [hunk 1]
```diff
@@ -47,6 +51,9 @@ public class Kinesis2Configuration implements Cloneable {
     @UriParam(description = "Amazon Kinesis client to use for all requests for this endpoint")
     @Metadata(label = "advanced", autowired = true)
     private KinesisClient amazonKinesisClient;
+    @UriParam(description = "Amazon Kinesis async client to use for all requests for this endpoint that require an async client")
```
Comment (oscerd) [inline components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java:54]: If it's specific to KCL is better to add the info to the description

**Review by oscerd — CHANGES_REQUESTED** [review]
One minor finding

File: components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java (hunk 1 above)
Comment (mikey-austin) [reply components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java:54]: No worries, I'll add a better description for this param.

Comment (oscerd) [reply components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java:54]: Thanks. I'm also curious about your use case, are you using this in production? What version? Did you see space for more improvements?

Comment (mikey-austin) [reply components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java]: I'm currently evaluating this for a use case in production. It has been working pretty well so far, but I am still yet to do some serious load testing, which will probably happen after New Years. One possible improvement I thought of was around potentially injecting a custom AWS credential provider implementation and make use of the library client instantiation. It was a more involved patch, so for now I went with the current approach which also works.

**Review by oscerd — APPROVED** [review]

**Comment by mikey-austin** [comment]
> /component-tests aws
> 
> **Result** ❌ The tests failed please [check the logs](https://github.com/apache/camel/actions/runs/12483491171)

The failing test (`org.apache.camel.component.aws2.s3.integration.S3CopyObjectCustomerKeyIT.sendIn`) in the java 17 build doesn't seem related to my change (unless I'm misreading the log output there).

**Comment by oscerd** [comment]
Thanks for the PR, I'm going to fully build to regenerate the DSL. Good to know about usage.

**Comment by mikey-austin** [comment]
Hi @oscerd will this change eventually be released on 4.9.x? I saw some recent releases on 4.8.x and I'm not sure about the process for new commits on main to make their way into a proper release.

**Comment by oscerd** [comment]
Hello @mikey-austin, this will be released on 4.10.0 by the end of February. It's not in 4.9.x which is a development version and will only have 4.9.0 released (released before your commit). 4.10.0 will be LTS.
